src/constants/screens.js

```diff
@@ -1,43 +1,60 @@
 /**
- * This file cannot be converted to TypeScript because it's used by the Tailwind config
+ * This file cannot be converted to TypeScript because it's referenced in the
+ * Tailwind config.
+ *
+ * @typedef {'2xl' | 'xl' | 'lg' | 'md' | 'sm' | 'xs'} Breakpoint
+ * @typedef {Exclude<Breakpoint, 'xs'>} RealBreakpoint
+ * @typedef {{ name: string, styles: { width: string, height: string }}} Viewport
  */
-/** @typedef {'2xl' | 'xl' | 'lg' | 'md' | 'sm' | 'xs'} Breakpoint */
 
 /**
- * Mapping of a breakpoint name to the lower-bound of its screen-width range
+ * mapping of breakpoint names to the lower-bound of their screen width range
+ *
+ * @type {Record<RealBreakpoint, number>}
  */
-const SCREEN_SIZES = new Map(
-  /** @type {const} */ ([
-    ['2xl', 1536],
-    ['xl', 1280],
-    ['lg', 1024],
-    ['md', 768],
-    ['sm', 640],
-  ])
-)
+const SCREEN_SIZES = {
+  '2xl': 1536,
+  xl: 1280,
+  lg: 1024,
+  md: 768,
+  sm: 640,
+}
 
-const viewportEntries = [
-  ...Array.from(SCREEN_SIZES),
-  /** @type {const} */ (['xs', 340]),
-].map(([key, val]) => {
-  return /** @type {const} */ ([
-    key,
-    {
-      name: `${key} (${val}px)`,
-      styles: { width: `${val}px`, height: '768px' },
-    },
-  ])
-})
+/**
+ * mapping of breakpoint names to the lower-bound of their screen width range,
+ * with the 'px' suffix attached; This is used by Tailwind.
+ */
+const SCREENS = /** @type {Record<RealBreakpoint, string>} */ (
+  Object.fromEntries(
+    Object.entries(SCREEN_SIZES)
+      .map(([key, val]) => [key, `${val}px`])
+      .reverse()
+  )
+)
 
-const VIEWPORTS =
-  /** @type {Record<Breakpoint, typeof viewportEntries[number][1]>} */ (
-    Object.fromEntries(viewportEntries)
+/**
+ * mapping of breakpoint names to viewport specifications; This is used by
+ * the viewports plugin of Storybook.
+ */
+const VIEWPORTS = /** @type {Record<Breakpoint, Viewport>} */ (
+  Object.fromEntries(
+    Object.entries(SCREEN_SIZES)
+      .concat([['xs', 340]])
+      .map(([key, val]) => [
+        key,
+        {
+          name: `${key} (${val}px)`,
+          styles: { width: `${val}px`, height: '768px' },
+        },
+      ])
   )
+)
 
 /**
- * This module is consumed by Nuxt and Tailwind config so it needs to use CJS modules
+ * This module is consumed by Nuxt and Tailwind config, so it needs to use CJS modules.
```

Comment:
Nuxt 3 expects no CJS modules. Can you think of a better way of having a single source of truth for screens for both Nuxt and Tailwind without using a CJS module? This is not a comment for this PR, just a note since you're touching these files.

Reply:
The main blocker here is that Tailwind doesn't support anything other than CJS. I don't see any way other than duplicating the constants in two places.